Element84/filmdrop-ui
# Move map logic out of search component and refactor related files

## Related Issue(s):

- #110 

## Proposed Changes:

1. move map logic out of search component
2. move API calls to be isolated into separate services under a services directory
3. refactor API calls for `GET` `/aggregations` and `/queryables` to be only called once on application initialization
4. app now closes popup on map click and no results under cursor
5. app now clears map selection on popup close
6. app now clears map selection on collection changed
7. move dataTimeRange to it's own component
8. fix styling in popup results component

**BONUS:**
-  add image placeholder to be used in results popup for scenes that don't include a thumbnail
- fix query params sent in error when not needed
- fix css z-order of modals
- fix date range 'invalid date' message and refactor end to be 'present' when no end date or present date set
- disable selection of 'Select One' placeholder text in Collection dropdown and update text to be 'Select Collection'

## TO TEST:

### Manual tests for new functionality:

**confirm aggregations and queryables only loaded on init**
- open browser dev tools and go to network tab
- reload app
- confirm calls for `/aggregations` & `/queryables` are made when app loads
- change collection
- confirm `/queryables` is no longer called on change

**popup close and map selection clear together**
- load app
- click search
- click on any cell in map
- confirm popup showing results loads
- confirm cell gets highlighted and image overlay loads on map
- click on map where there is no data
- confirm popup results and map selection both clear
- click on any cell in map again
- click close 'X' button in top left of popup results
- confirm popup results and map selection both clear

**confirm result scenes/grid clear, selection clears, and popup closes on:
new search button click OR on collection change OR on autosearch toggle OR on mosaic toggle**
- load app
- zoom in to scene mode or grid code showing
- click search button
- confirm results show on map
- click scene(s)
- confirm popup shows
- confirm orange highlight for click scenes shows
- confirm image overlay loads (if collection supports it)
- click search button again
- confirm popup closes
- confirm selection clears
- confirm image overlay clears
- click scene(s) again
- click map outside of scenes
- confirm popup closes
- confirm selection clears
- confirm image overlay clears
- click scene(s) again
- change collection
- confirm popup closes
- confirm selection clears
- confirm image overlay clears
- click scene(s) again
- toggle auto search on
- confirm popup closes
- confirm selection clears
- confirm image overlay clears
- toggle auto search off again
- click scene(s) again
- toggle mosaic view mode
- confirm popup closes
- confirm selection clears
- confirm image overlay clears

**image placeholder when no thumbnail**
- load app
- change collection to 'Copernicus DEM GLO-90'
- click search
- zoom in until grid or scene view shows
- click grid or scene
- confirm popup shows 'Thumbnail Not Available' image placeholder in popup result

**modals z-index fixed**
- load app
- click 'launch your own filmdrop UI' button
- confirm modal loads on top of the rest of the app

**fix query param sent in error**
- load app
- confirm 'Sentinel-2 Level 2A' is the currently selected collection
- zoom in to view scene mode
- open browser dev tools, click network tab
- click search
- look at request URL sent
- confirm request includes `query` param
- change collection to 'Copernicus DEM GLO-90'
- click search button again
- look at request URL sent
- confirm request does not include `query` param

**no more invalid date message on load**
- load app
- hover cursor over info icon next to date range title text
- confirm start date shows a valid date
- confirm end date says 'present' 
- change collection to 'Copernicus DEM GLO-90'
- confirm start date shows a valid date
- confirm end date shows a valid date
- change collection back to 'Sentinel-2 Level 2A'
- confirm start date shows a valid date
- confirm end date says 'present' 

**disabled select placeholder text**
- load app
- click Collection select list
- confirm placeholder text at top of list is 'Select Collection'
- confirm placeholder text at top of list is not selectable 

### Manual tests to confirm no regressions:

> **_NOTE:_**  These changes should not have introduces any regressions into existing application functionality. Reviewer should confirm that existing application works as expected. Existing features were too numerous to reasonably list as manual PR tests here, reach out if manual list of functionality test cases desired.  

**test with additional config**
- get alternative `config.js` settings other than demo dev config (example config exists currently only in a private repo, PM if need help locating) 
- comment out old and paste in same file or replace `config.js` file completely with alternate
- restart local app using new config 
- confirm app loads with non demo config

## PR Checklist:
